trusted-services.mk

```diff
@@ -97,3 +97,40 @@ linux-arm-ffa-user-clean:
 # static carveout. If changed, run "make linux-defconfig-clean" to take effect.
 LINUX_DEFCONFIG_COMMON_FILES += $(CURDIR)/kconfigs/fvp_trusted-services.conf
 endif
+
+################################################################################
+# Trusted Services test applications
+################################################################################
+.PHONY: ffa-test-all ffa-test-all-clean ffa-test-all-realclean
+all: ffa-test-all
+
+ffa-test-all-realclean:
+	rm -rf $(TS_INSTALL_PREFIX)/arm-linux
+
+ifneq ($(COMPILE_NS_USER),64)
+$(error Trusted Services test apps only support AArch64)
+endif
+
+# Helper macro to build and install Trusted Services test applications.
+# Invokes CMake to configure, and make to build and install the apps.
+define build-ts-app
+.PHONY: ffa-$1
+ffa-$1:
+	CROSS_COMPILE=$(subst $(CCACHE),,$(CROSS_COMPILE_NS_USER)) cmake -G"Unix Makefiles" \
```

Comment:
How come `$(CCACHE)` needs to be removed?

Reply:
CMake has another mechanism to pass ccache and we only expect the compiler prefix in this variable. Please see https://github.com/OP-TEE/build/pull/571#discussion_r904930894 for details.